X-wing/resources/x-wing/components/upgrade-card/upgrade-card.js

```diff
@@ -389,50 +427,57 @@ function paintCardFaceComponents( g, diy, sheet, side) {
 			sheet.drawOutlinedTitle( g, range, R('lower-range'), Xwing.numberFont, 8, 1, Color.WHITE, Color.BLACK, sheet.ALIGN_CENTER, true);
 			nameBox.draw( g, R('short-name') );
 		}
-	} else {
-		if ( energyLimit == '-' ) {
-			sheet.paintImage( g, 'upgrade-normal-template', 0, 0 );
-			nameBox.draw( g, R('name') );
-		} else {
+	} else { // not secondary weapon
+		if ( energyLimit == '-'){ // not an epic card ( w/ energy)
+			if (style == 'full' ) { // full art: no overlay but alt region
+				nameBox.draw( g, R('name-alt') ); // moved region
+			} else { // regular style
+				sheet.paintImage( g, 'upgrade-normal-template', 0, 0 );
+				nameBox.draw( g, R('name') );
+			}
+		} else if (energyLimit != '-') { // energy
 			sheet.paintImage( g, 'upgrade-energy-template', 0, 0 );
 			sheet.drawOutlinedTitle( g, energyLimit, R( 'upper-attribute' ), Xwing.numberFont, 14, 1, Xwing.getColor('energy'), Color.BLACK, sheet.ALIGN_CENTER, true);
 			nameBox.draw( g, R('short-name') );
 		}
 	}
-		
+
 	// Determine Text Area
-	a = 0;
-	b = 0;
-	c = 0;
-	d = 0;
-	e = 45;
-	if( secondaryWeapon ) {
-		if( energyLimit == '-' ) {
-			a = 62;
-			b = 481;
-			c = 0;
-		} else {
-			a = 79;
-			b = 493;
-			c = 62;			
+	if ( style != 'full' ){
+		a = 0;
+		b = 0;
+		c = 0;
+		d = 0;
+		e = 45;
+		if( secondaryWeapon ) {
+			if( energyLimit == '-' ) {
+				a = 62;
+				b = 481;
+				c = 0;
+			} else {
+				a = 79;
+				b = 493;
+				c = 62;
+			}
 		}
-	}
-	if( $UpgradeType != 'modification' && $UpgradeType != 'title' ) {
-		if(  $$DoubleIcon.yesNo ) {
-			d = 110;
-			e = 125;
-		} else {
-			d = 30;
+		if( $UpgradeType != 'title' ) {
```

Comment:
What happened to the " $UpgradeType != 'modification'" statement. Modifications do not have icons either...

Reply:
Ok, I see. I think they are being lazy and don't want to add graphics for a iconless template :)

The same is probably true for titles as well, should we ever see one of those as a promo so it should not be different from modifications.

As we don't want to mess with the original templates, a check needs to be added here. If it is promo then treat titles and modifications as any other icons, but otherwise we keep it as it were (i.e. both title and modification).